public/script.js

```diff
@@ -0,0 +1,249 @@
+  artists.forEach(artist => {
+    artistSection.querySelector('.items').insertAdjacentElement('beforeend', objectToHtmlElement(artist));
+  });
+  return artistSection;
+}
+
+/**
+*  Функция для создания раздела с треками
+* @param {object[]} tracks - массив треков
+* @returns раздел, содержащий ссылки на треки
+*/
+function tracksToHtml(tracks) {
+  let trackSection = createSection().cloneNode(true);
+  trackSection.querySelector('.section__header').textContent = 'Треки';
+  trackSection.className = 'track-section';
+  trackSection.querySelector('.items').className = 'tracks';
+  tracks.forEach(track => {
+    trackSection.querySelector('.tracks').insertAdjacentHTML('beforeend', 
+    `<a href='${track.external_urls.spotify}' 
+     class='track link'>${artistsToString(track.artists)} - ${track.name}</a>`);
+  });
+  return trackSection;
+}
+
+/** 
+ * Функция для вывода результатов поиска на страницу 
+ * @param {object} obj - объект, содержащий результаты поиска
+*/
+function searchResults(obj) {
+  content.innerHTML = "";
+  const albums = obj.albums.items;
+  const artists = obj.artists.items;
+  const tracks = obj.tracks.items;
+  if(albums.length == 0 && artists.length == 0 && tracks.length == 0)
+    content.insertAdjacentHTML('afterbegin', '<div class="not-found">По вашему запросу ничего не найдено</div>');
+  else {
+    if(albums.length != 0)
+      content.insertAdjacentElement('beforeend', albumsToHtmlElement(albums));
+    if(artists.length != 0)
+      content.insertAdjacentElement('beforeend', artistsToHtmlElement(artists));
+    if(tracks.length != 0)
+      content.insertAdjacentElement('beforeend', tracksToHtml(tracks));
+  }
+}
+
+/** 
+ * Функция для вывода категорий плейлистов на страницу 
+ * Внутри функции создается массив категорий плейлистов
+ * Если второй параметр не передан, то выводятся все категории, начиная со start
+ * @param {number} start - индекс в массиве категорий плейлистов, с которого начинается вывод
+ * @param {number} end - индекс в массиве категорий плейлистов, до которого происходит вывод
+ */
+async function showPlaylists(start, end) {
+  const token = await getToken();
+  const categories = await getCategories(token);
+  if(typeof(end) == 'undefined')
+    end = categories.length;
+  for(let i = start; i < end; i++) {
+    const playlists = await getPlaylists(token, categories[i].id);
+    if(typeof(playlists) != 'undefined' && playlists.length > 0) 
+      content.insertAdjacentElement('beforeend', playlistsToHtmlElement(playlists, categories[i].name));
+  }
+}
+
+/**
+ * Функция предназначена для удаления со страницы всех категорий плейлистов, кроме первых пяти
+ */
+function hidePlaylists() {
+  const sections = document.querySelectorAll('.section');
+  for(let i = 5; i < sections.length; i++) {
+    sections[i].remove();
+  }
+}
+
+content.addEventListener('click', (event) => {
+  if(event.target.className == 'show-all') {
```

Comment:
стоит написать так
```suggestion
  if(event.target.className === 'show-all') {
```

Также если вдруг добавятся другие классы у элемента, то этот код перестанет срабатывать. Это не замечание, просто на всякий случай